recipes/libmpdclient/all/conanfile.py

```diff
@@ -0,0 +1,111 @@
+    license = "BSD-3-Clause"
+    topics = ("music", "music-player-demon", "sound")
+    homepage = "https://www.musicpd.org/libs/libmpdclient"
+    url = "https://github.com/conan-io/conan-center-index"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "default_socket": ["ANY"],
+        "default_host": ["ANY"],
+        "default_port": ["ANY"],
+        "tcp": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "default_socket": "/var/run/mpd/socket",
+        "default_host": "localhost",
+        "default_port": "6600",
+        "tcp": True,
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            del self.options.default_socket
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def validate(self):
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support build of shared lib with msvc")
+
+    def build_requirements(self):
+        self.tool_requires("meson/1.2.1")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+        tc = MesonToolchain(self)
+        if self.settings.os != "Windows":
+            tc.project_options["default_socket"] = str(self.options.default_socket)
+        tc.project_options["default_host"] = str(self.options.default_host)
+        tc.project_options["default_port"] = str(self.options.default_port)
+        tc.project_options["tcp"] = self.options.tcp
+        tc.project_options["documentation"] = False
+        tc.project_options["test"] = False
+        tc.generate()
+
+    def build(self):
+        meson = Meson(self)
+        meson.configure()
+        meson.build()
+
+    def package(self):
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        meson = Meson(self)
+        meson.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        fix_apple_shared_install_name(self)
+        fix_msvc_libname(self)
+
+    def package_info(self):
+        self.cpp_info.set_property("pkg_config_name", "libmpdclient")
+        self.cpp_info.libs = ["mpdclient"]
+        if self.settings.os == "Windows":
+            self.cpp_info.system_libs.append("ws2_32")
+
+def fix_msvc_libname(conanfile, remove_lib_prefix=True):
```

Comment:
You might have had this asked before but I couldnt find anything in the few PRs I checked:

Is there any strong reason besides conventions for you to prefer doing it this way instead of conditionally setting `self.cpp_info.libs`? I'm not against this way either, but it's true that I feel it would leave the code simpler to understand and be more "true", as it leaves the filename unchanged to what might otherwise be found in the wild

Reply:
It would be another solution yes. Currently in all meson recipes, lib file is renamed for msvc, so it's consistent. Actually I was hoping that conan would provide this helper function (https://github.com/conan-io/conan/issues/11866) in the near future.

Conditional lib names in package_info() tend to be fragile (many issues in CCI recipes with clang variant on Windows), I think the current workaround is more robust.